### PR TITLE
fix: remove py.typed since package fails mypy check

### PR DESCRIPTION
**Pull request by tswast**
Reverts googleapis/python-bigquery#976

Context: https://issues.apache.org/jira/browse/BEAM-12975

### CODE REVIEW

**Comment by tswast** [comment]
Will check the library with `mypy` before reverting.

**Review by dandhlee — APPROVED** [review]

**Comment by plamut** [comment]
Do we know what exactly caused that error? Is it an issue with some of our annotations, or does `mypy` behave differently from `pytype`?

It would be good to fix this while still retaining `py.typed`, although removing it as a short term fix to unblock maintainers of other is fine.

**Pull request merged**
